addons/sys_logistics/fnc_setObjectCargo.sqf

```diff
@@ -70,19 +72,35 @@ _typesWeapons = [[_cargoW,"WeaponCargo"],[_cargoM,"MagazineCargo"],[_cargoI,"Ite
 
     _content = _x select 0;
     _operation = _x select 1;
-    _current = call (compile format["get%1 _input",_operation]);
 
-    if !(str(_content) == str(_current)) then {
+    _currenttemp = [{getWeaponCargo _input}, {getMagazineCargo _input},{getItemCargo _input}];
+    _current = call (_currenttemp select (["weaponcargo","magazinecargo","itemcargo"] find (tolower _operation)));
 
-        if (count (_current select 0) > 0) then {call (compile format["clear%1%2 _input",_operation,_global])};
+    if !(_content isEqualTo _current) then {
+
+        if (count (_current select 0) > 0) then {
+            
+            if (_global == "Global") then {
```

Comment:
`private _actions = if (globa....) then { [arr] } else { [arr] }`

if _actions hasn't been delcared before. Then it will be undefined in L88 in the current state.